proxy/grpcproxy/watch.go

```diff
@@ -229,11 +229,14 @@ func (wps *watchProxyStream) recvLoop() error {
 		case *pb.WatchRequest_CreateRequest:
 			cr := uv.CreateRequest
 
-			if err = wps.checkPermissionForWatch(cr.Key, cr.RangeEnd); err != nil && err == rpctypes.ErrPermissionDenied {
-				// Return WatchResponse which is caused by permission checking if and only if
-				// the error is permission denied. For other errors (e.g. timeout or connection closed),
-				// the permission checking mechanism should do nothing for preserving error code.
-				wps.watchCh <- &pb.WatchResponse{Header: &pb.ResponseHeader{}, WatchId: -1, Created: true, Canceled: true}
+			if err := wps.checkPermissionForWatch(cr.Key, cr.RangeEnd); err != nil {
```

Comment:
if the error type is not permission denied, the watcher error should still be handled somewhere else, no?

Reply:
we probably should still keep line 232, and only change line 236 to include the permission denied error.

Reply:
> How was a timeout, connection closed supposed to be handled if the code swallows the error and the proxy silently never creates the watch?

From the comment, it seems that the error should be handled somewhere else. I looked at the code again, it seems that handling the other types of errors (due to the failure of permission checking here) is OK.